src/parser/vast_parser.js

```diff
@@ -108,13 +108,14 @@ export class VASTParser extends EventEmitter {
    * Fetches a VAST document for the given url.
    * Returns a Promise which resolves,rejects according to the result of the request.
    * @param  {String} url - The url to request the VAST document.
-   * @param {Number} wrapperDepth - how many times the current url has been wrapped
-   * @param {String} previousUrl - url of the previous VAST
+   * @param {Number} wrapperDepth - How many times the current url has been wrapped.
+   * @param {String} previousUrl - Url of the previous VAST.
+   * @param {Object} ad - Previously parsed ad node (Wrapper) related to this fetching.
    * @emits  VASTParser#VAST-resolving
    * @emits  VASTParser#VAST-resolved
    * @return {Promise}
    */
-  fetchVAST(url, wrapperDepth = 0, previousUrl = null) {
+  fetchVAST(url, wrapperDepth = 0, previousUrl = null, ad = null) {
```

Comment:
What do you think about renaming `ad` to `wrapperAd` to make it more explicit (especially on the markdown documentation)
```suggestion
  fetchVAST(url, wrapperDepth = 0, previousUrl = null, wrapperAd = null) {
```